vision/src/autogluon/vision/image_prediction/image_prediction.py

```diff
@@ -144,13 +160,23 @@ def fit(self,
                 'time_limits': 2147483647 if time_limit is None else max(1, time_limit),
                 'search_strategy': search_strategy,
                 }
-        if nthreads_per_trial is not None:
-            config['nthreads_per_trial'] = nthreads_per_trial
-        if ngpus_per_trial is not None:
-            config['ngpus_per_trial'] = ngpus_per_trial
-        if dist_ip_addrs is not None:
-            config['dist_ip_addrs'] = dist_ip_addrs
+        if num_cpus is not None:
+            config['nthreads_per_trial'] = num_cpus
+        if num_gpus is not None:
+            config['ngpus_per_trial'] = num_gpus
         if isinstance(hyperparameters, dict):
+            if 'batch_size' in hyperparameters:
+                bs = hyperparameters['batch_size']
+                if num_gpus is not None and num_gpus > 1 and bs > 64:
+                    # using gpus, check batch size vs. available gpu memory
+                    free_gpu_memory = get_gpu_free_memory()
+                    if not free_gpu_memory:
+                        warnings.warn('Unable to detect free GPU memory, we are unable to verify your batch_size setting.')
```

Comment:
"your batch_size setting" -> "whether your data mini-batches will fit on the GPU for the specified batch_size"